packages/govern-console/src/containers/CreateDao/CreateDaoCollateral.tsx

```diff
@@ -96,321 +95,314 @@ const CreateDaoCollateral: React.FC<{
-                  onClick={() => {
-                    append({ value: '' });
-                  }}
-                />
-              </div>
             )}
-
-            <Split
-              width={'100%'}
-              primary={
-                <Button
-                  size={'large'}
-                  mode={'secondary'}
-                  onClick={() => {
-                    setActiveStep(CreateDaoSteps.Config);
+          />
+        </div>
+        {watch('isAnyAddress') ? (
+          <Info mode={'warning'} title={''}>
+            If you select ”Any Address”, then everybody can schedule executions in your DAO. Please
+            be sure you understand the impact of this selection.
+          </Info>
+        ) : (
+          <div>
+            {fields.map((item: any, index: any) => {
+              return (
+                <div
+                  key={item.id}
+                  style={{
+                    display: 'flex',
+                    flexDirection: 'column',
                   }}
-                  icon={<IconArrowLeft />}
-                  label={'back'}
-                  display={'all'}
-                />
-              }
-              secondary={
-                <Button wide size={'large'} mode={'secondary'} onClick={moveToNextStep}>
-                  Next Step
-                </Button>
-              }
+                >
+                  <div
+                    style={{
+                      display: 'flex',
+                      flexDirection: 'row',
+                      gap: 16,
+                      marginTop: 8,
+                    }}
+                  >
+                    <Controller
+                      name={`executionAddressList[${index}].value`}
+                      control={control}
+                      defaultValue={item.value}
+                      rules={{ required: 'This is required.' }}
+                      render={({ field: { onChange, value }, fieldState: { error } }) => (
+                        <TextInput
+                          wide
+                          value={value}
+                          onChange={onChange}
+                          status={!!error ? 'error' : 'normal'}
+                          error={error ? error.message : null}
+                        />
+                      )}
+                    />
+                    <Button
+                      mode={'secondary'}
+                      size={'large'}
+                      disabled={fields.length === 1}
+                      icon={<IconMinus />}
+                      onClick={() => {
+                        remove(index);
+                      }}
+                    />
+                  </div>
+                </div>
+              );
+            })}
+            <Button
+              wide={false}
+              mode={'secondary'}
+              size={'large'}
+              disabled={fields.length >= 10}
+              label={'Add new address'}
```

Comment:
why are we hardcoding `fields.length >= 10`??

Reply:
because max addresses allowed is 10 according to https://github.com/aragon/govern/blob/develop/packages/govern-create/contracts/GovernBaseFactory.sol#L20
but i will move it to constant anyway